### ISSUE TITLE
Audit: Mitigate reflow / content layout shifts

### ISSUE THREAD

**Issue by Malvoz** [GoogleChrome/lighthouse#10085]
(This feature request may be considered part of https://github.com/GoogleChrome/lighthouse/issues/5287, but doesn't entail as much, as a single audit.)

The [`intrinsicsize`](https://github.com/WICG/intrinsicsize-attribute) attribute was proposed to allow developers to have images maintain aspect ratio, proportional to the width of the screen, without causing a user visible reflow.
The attribute proposal was superseded by the [new UA-defined styles](https://github.com/WICG/intrinsicsize-attribute/issues/16) ([Chrome status](https://chromestatus.com/feature/5695266130755584)):
```CSS
img, video {
    aspect-ratio: attr(width) / attr(height);
} 
```

The WHATWG HTML spec is also to [recommend developers](https://github.com/whatwg/html/commit/c04c0be3afe4e0039c55e16eae621e5fa4920fbd#diff-36cd38f49b9afa08222c0dc9ebfe35ebR26317-R26320) to set `width` and `height` attributes (specifically for images, when lazy-loading), however most developers usually don't seek information in the spec and so an audit would increase the likelyhood that developers gets exposed to this recommendation.

As a side note, I think it's important that developers don't disable [scroll anchoring](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-anchor/Guide_to_scroll_anchoring) by setting (or inheriting) [`overflow-anchor: none`](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-anchor).

Thanks for taking your time.

**Comment by addyosmani** [comment]
Yeah. Native image lazy-loading also recommends developers specify a `width` and `height` to avoid relayout:

![image](https://user-images.githubusercontent.com/110953/70497228-51b74200-1ac7-11ea-870b-51ace22963d1.png)

I think specifying dimensions would be useful for us to factor into either a new or existing audit.

**Comment by Malvoz** [comment]
@paullewis worked on pretty much the same thing in https://github.com/GoogleChrome/lighthouse/pull/511 (which is inactive and closed) but there's https://github.com/GoogleChrome/lighthouse/issues/802 by @samccone, closing this.

**Comment by paulirish** [comment]
I prefer the modern context here vs the old issue so let's dupe 511 against this. ;)

**Comment by pawelurbanski** [comment]
It would be useful to provide information how to properly apply width and height attributes. The case is important for the picture element and the img element required by picture.
There can be possible association of width and height specified at the img tag as the dimensions the browser should display the image. Making more clear what is the relation to media queries and sizes attribute would be useful.
For example:
The page has an image that is: 1500x1500 pixels. It is ment to be displayed on mobile devices of max-width: 400px for most mobiles use screens at 4 DPI.
What would be a proper markup using width and height to hint the ratio using the picture element with a source element pointing at this image with a media query.

**Comment by Malvoz** [comment]
@pawelurbanski there's a proposal to introduce `width` and `height` for `<source>` in https://github.com/whatwg/html/issues/4968. Until we (hopefully) have that; for `<img>` I'd personally specify the intrinsic size of the image file that is to be displayed on mobile/small devices since they generally perform slower than larger devices.

**Comment by pawelurbanski** [comment]
Such explanation would be very useful, for example:
“Specify width and height attribute on a source element.”
I understand that it is enough to provide the width and height values that correspond to the image listed in the srcset attribute, which is the smallest for example, as long as it represents the aspect ratio of all the other images of different sizes.

**Comment by paulirish** [comment]
There are now a few audits implemented that target understanding/mitigating Layout Shifts.

--------

There's also now a new webplatform feature, **Font Metrics Override Descriptors**, that has potential to help more. It's shipping in Chrome 87.
https://gist.github.com/xiaochengh/da1fa52648d6184fd8022d7134c168c1#use-case-reduce-layout-shifting-caused-by-web-fonts

**Comment by paulirish** [comment]
aspect-ratio is now handled in https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/layout/layout_image.cc;l=375-386;drc=02c21516ef901e09640e417bde686ba469d7438a which is used in the `LayoutImageUnsized` trace event . Those results are then shown in the "Layout shift culprits" insight

and for the record, aspect-ratio was handled in lighthouse way back in  https://github.com/GoogleChrome/lighthouse/pull/12377
cc https://github.com/GoogleChrome/lighthouse/issues/12218

--------

> Font Metrics Override Descriptors

i added this to  go/lighthouse-rules which will keep track of it in the backlog.
